### PR TITLE
Revamp release actions

### PR DESCRIPTION
**Pull request by benwoo1110**
- Update DBO upload to use https://github.com/benwoo1110/dbo-upload-action/
- Create manual upload function for Modrinth
- Release DBO and Modrinth on push
- Release DBO and Modrinth on promote release
- Use new semantic action https://github.com/benwoo1110/semantic-release-action

### CODE REVIEW

File: .github/workflows/dbo_upload.yml [hunk 1]
```diff
@@ -19,40 +27,31 @@ jobs:
...
+              echo "Replacing prerelease version scheme with SNAPSHOT"
+              echo "RELEASE_TYPE=beta" >> $GITHUB_ENV
+          else
+              echo "Using release version scheme"
+              echo "RELEASE_TYPE=release" >> $GITHUB_ENV
+          fi
 
       - name: Upload to dev.bukkit.org
-        uses: Multiverse/http-request-action@debug
+        uses: benwoo1110/dbo-upload-action@v0.1.0
```
Comment (dumptruckman) [inline .github/workflows/dbo_upload.yml:41]: image

**Comment by dumptruckman** [comment]
What is the point of having separate dbo and modrinth manual workflows if the release and promote release both do this?

File: .github/workflows/promote_release.yml [hunk 2]
```diff
@@ -47,62 +39,71 @@ jobs:
...
-        run: |
-          echo "Removing prerelease tag from version"
-          echo "VERSION=$(echo ${{ steps.get-release.outputs.tag_name }} | sed -E 's/-pre.*//')" >> $GITHUB_ENV
-
-      - name: Build
+      - name: Build and test
         uses: gradle/gradle-build-action@v2
         with:
-          arguments: clean build -x test -x checkstyleMain -x checkstyleTest
+          arguments: clean build -x assemble -x shadowJar
```
Comment (dumptruckman) [inline .github/workflows/promote_release.yml:45]: Why change this?

Comment (benwoo1110) [reply .github/workflows/promote_release.yml:45]: I think we should run the tests before promoting the release? Just incase something uncaught

**Review by dumptruckman — APPROVED** [review]

**Pull request merged**